nemo_curator/modules/fuzzy_dedup.py

```diff
@@ -1410,49 +1461,60 @@ def _run_connected_components(
         deduped_parsed_id_path,
         output_path,
     ):
-        Comms.initialize(p2p=True)
-        df = dask_cudf.read_parquet(
-            deduped_encoded_jaccard_path, blocksize="1GB", aggregate_files=True
-        )
-        df = df[df["jaccard"] == 1].reset_index(drop=True)
+        t0 = time.time()
+        with performance_report_if(
+            self.profile_dir,
+            f"connected-components-run-{datetime.now().strftime('%Y%m%d_%H%M%S')}.html",
+        ):
+
+            Comms.initialize(p2p=True)
+            df = dask_cudf.read_parquet(
+                deduped_encoded_jaccard_path, blocksize="1GB", aggregate_files=True
+            )
+            df = df[df["jaccard"] == 1].reset_index(drop=True)
 
-        labels_df = dask_cudf.read_parquet(deduped_parsed_id_path)
-        num_nodes = len(labels_df)
-        self_edge_df = labels_df[["uid"]].rename(columns={"uid": self.left_id})
-        self_edge_df[self.right_id] = self_edge_df[self.left_id]
+            labels_df = dask_cudf.read_parquet(deduped_parsed_id_path)
+            num_nodes = len(labels_df)
+            self_edge_df = labels_df[["uid"]].rename(columns={"uid": self.left_id})
+            self_edge_df[self.right_id] = self_edge_df[self.left_id]
 
-        df = df[[self.left_id, self.right_id]].astype(np.int64)
-        df = dask_cudf.concat([df, self_edge_df])
+            df = df[[self.left_id, self.right_id]].astype(np.int64)
+            df = dask_cudf.concat([df, self_edge_df])
 
-        G = MultiGraph(directed=False)
-        G.from_dask_cudf_edgelist(
-            df, source=self.left_id, destination=self.right_id, renumber=False
-        )
-        result = dcg.weakly_connected_components(G)
-        del G
-        max_partitions = min(32, result.npartitions)
-        n_components = len(result[["labels"]].drop_duplicates(split_out=max_partitions))
-        num_labels = len(result)
-        print("# of groups", n_components)
-        print("# of docs removed", num_labels - n_components)
-        labels_df = labels_df.merge(
-            result, left_on=["uid"], right_on=["vertex"], how="inner"
-        )
-        id_columns = (
-            ["dataset_id", "doc_id"] if self.convert_str_ids else [self.id_column]
+            G = MultiGraph(directed=False)
+            G.from_dask_cudf_edgelist(
+                df, source=self.left_id, destination=self.right_id, renumber=False
+            )
+            result = dcg.weakly_connected_components(G)
+            del G
+            max_partitions = min(32, result.npartitions)
+            n_components = len(
+                result[["labels"]].drop_duplicates(split_out=max_partitions)
+            )
+            num_labels = len(result)
+            print("# of groups", n_components)
+            print("# of docs removed", num_labels - n_components)
```

Comment:
@ayushdg do you have recommendation on logging these using logger, and if you have preference of a certain format. In other places we are following the format
> Time taken for **{NAME}** = **{TIME}** and output written at **{PATH}**

Reply:
I don't have a preference but I agree we should log this in the logger with some standard prefix like `Result ...`